lib/smb2/file.rb

```diff
@@ -119,4 +165,51 @@ def size
     info.end_of_file
   end
 
+  # Write the entire contents of `data`, starting at `offset` bytes from the
+  # beginning of the file.
+  #
+  # @param data [String] what to write
+  # @param offset [Fixnum] where in the file to start writing
+  # @return [Fixnum] number of bytes written. This may be less than the length
```

Comment:
Is there anyway for the caller to get the error?  If so, that should be documented.